File: ui-android-library/src/main/java/com/xyfindables/ui/dialogs/XYThrobberDialog.java
package com.xyfindables.ui.dialogs;

/*
 Created by arietrouw on 12/27/17.
 */

import android.app.Activity;
import android.app.Dialog;
import android.graphics.drawable.ColorDrawable;
import android.view.View;
import android.view.Window;

import com.xyfindables.ui.R;
import com.xyfindables.ui.XYBaseActivity;

import static com.xyfindables.core.XYBase.logInfo;

public class XYThrobberDialog extends Dialog {

    final private static String TAG = XYThrobberDialog.class.getSimpleName();

    private Activity _activity;

    public XYThrobberDialog(Activity activity) {
        super(activity);
        _activity = activity;
        setContentView(R.layout.dialog_throbber);
        Window window = this.getWindow();
        if (window != null) {
            this.getWindow().setBackgroundDrawable(new ColorDrawable(android.graphics.Color.TRANSPARENT));

            int dividerId = this.getContext().getResources().getIdentifier("android:id/titleDivider", null, null);
            View divider = this.findViewById(dividerId);
            if (divider != null) {
                divider.setVisibility(View.GONE);
            }
        }
    }

    public void show() {
        logInfo(TAG, "showThrobber");
        XYBaseActivity.RunOnUIThread(new Runnable() {
            @Override
            public void run() {
                if (_activity.isFinishing()) {
                    XYThrobberDialog.super.dismiss();
                } else {
                    XYThrobberDialog.super.show();
                }
            }
        });
    }

    public void hide() {
        logInfo(TAG, "showThrobber");
        XYBaseActivity.RunOnUIThread(new Runnable() {
            @Override
            public void run() {
                XYThrobberDialog.super.dismiss();
            }
        });
    }
}

File: ui-android-library/src/main/java/com/xyfindables/ui/dialogs/XYThrobberDialog.kt
package com.xyfindables.ui.dialogs

/*
 Created by arietrouw on 12/27/17.
 */

import android.app.Dialog
import android.content.Context
import android.graphics.drawable.ColorDrawable
import android.os.Bundle
import android.view.View

import com.xyfindables.ui.R
import com.xyfindables.ui.ui

class XYThrobberDialog(context: Context) : Dialog(context) {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.dialog_throbber)
        val window = this.window
        if (window != null) {
            this.window!!.setBackgroundDrawable(ColorDrawable(android.graphics.Color.TRANSPARENT))

            val dividerId = this.context.resources.getIdentifier("android:id/titleDivider", null, null)
            val divider = this.findViewById<View>(dividerId)
            if (divider != null) {
                divider.visibility = View.GONE
            }
        }
    }

    override fun show() {
        ui {
            super.show()
        }
    }

    override fun hide() {
        ui {
            super.hide()
        }
    }

    companion object {

        private val TAG = XYThrobberDialog::class.java.simpleName
    }
}
